apis/mutations/unversioned/value.go

```diff
@@ -35,34 +36,50 @@ type AssignField struct {
 
 	// FromMetadata assigns a value from the specified metadata field.
 	FromMetadata *FromMetadata `json:"fromMetadata,omitempty"`
+
+	// ExternalData describes the external data source to be used for mutation.
+	ExternalData *types.ExternalData `json:"externalData,omitempty"`
 }
 
-func (a *AssignField) GetValue(metadata types.MetadataGetter) (interface{}, error) {
+func (a *AssignField) GetValue(mutable *types.Mutable) (interface{}, error) {
 	if a == nil {
 		return nil, fmt.Errorf("assign is nil: %w", ErrInvalidAssignField)
 	}
 	if a.FromMetadata != nil {
-		return a.FromMetadata.GetValue(metadata)
+		return a.FromMetadata.GetValue(mutable.Object)
+	}
+	if a.Value != nil {
```

Comment:
The order of these if statements implies precedence if users provided more than one of `externalData`, `value` or `fromMetadata`. Do we allow multiple? if not, should we validate?

Reply:
I see the validation below to ensure only 1 is allowed.

Reply:
Those three fields are mutually exclusively and validated by `Validate()` so we can safely assume that we get the value as soon as we find a non-nil field (one of externalData, value or fromMetadata).